pym/bob/cmds/misc.py

```diff
@@ -131,6 +132,57 @@ def __init__(self, default, *args, **kwargs):
     def __missing__(self, key):
         return self.__default
 
+def doQueryMeta(argv, bobRoot):
+    parser = argparse.ArgumentParser(prog="bob query-meta",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description="""Query meta information of packages.""")
+    parser.add_argument('package', help="(Sub-)package to query")
+    parser.add_argument('-D', default=[], action='append', dest="defines",
+        help="Override default environment variable")
+    parser.add_argument('-c', dest="configFile", default=[], action='append',
+        help="Use config File")
+    parser.add_argument('-r', '--recursive', default=False, action='store_true',
+                        help="Recursively display dependencies")
+    parser.add_argument('-f', '--filter', default=[], action='append',
+                        help="Show only metaVars matched by filter.")
+    args = parser.parse_args(argv)
+
+    defines = {}
+    for define in args.defines:
+        d = define.split("=")
+        if len(d) == 1:
+            defines[d[0]] = ""
+        elif len(d) == 2:
+            defines[d[0]] = d[1]
+        else:
+            parser.error("Malformed define: "+define)
+
+    _filter = []
+    for f in args.filter:
+        _filter.append(re.compile(f))
+
+    recipes = RecipeSet()
+    recipes.setConfigFiles(args.configFile)
+    recipes.parse()
+    rootPackages = recipes.generatePackages(lambda s,m: "unused", defines)
+    package = walkPackagePath(rootPackages, args.package)
+
+    def showPackage(package, recurse, done=set()):
+        # show recipes only once for each checkout variant
+        key = (package.getRecipe().getName())
```

Comment:
Key should be `(package.getName(), package.getPackageStep().getVariantId())`. Currently it does not match the comment above.